spec/models/solidus_volume_pricing/pricer_spec.rb

```diff
@@ -0,0 +1,280 @@
+require 'spec_helper'
+
+RSpec.shared_examples 'having the variant price' do
+  it 'uses the variants price' do
+    is_expected.to eq('$10.00')
+  end
+end
+
+RSpec.describe SolidusVolumePricing::Pricer do
+  it 'inherits from default variant pricer' do
+    expect(described_class < Spree::Variant::PriceSelector).to be(true)
+  end
+
+  it 'has SolidusVolumePricing::PricingOptions as pricing options class' do
+    expect(described_class.pricing_options_class).to eq(SolidusVolumePricing::PricingOptions)
+  end
+
+  describe '#price_for' do
+    let(:variant) { create(:variant, price: 10) }
+    let(:role) { create(:role) }
+    let(:other_role) { create(:role) }
+    let(:user) { create(:user) }
+    let(:quantity) { 1 }
+
+    let(:pricing_options) do
+      SolidusVolumePricing::PricingOptions.new(quantity: quantity)
+    end
+
+    subject(:price_for) do
+      SolidusVolumePricing::Pricer.new(variant).price_for(pricing_options).to_s
+    end
+
+    before do
+      Spree::Config.volume_pricing_role = role.name
+    end
+
+    context 'discount_type = price' do
```

Comment:
prefer a description that chains well when tests are ran using `--format documentation`.

Currently it would read as
```
#price_for
  discount_type = price
    when quantity does not match the range
```

Instead it would be more clear with "when the discount_type is the same as the price"